tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,288 @@
+"""Test class for AzureRM Compute Resource
+
+:Requirement: ComputeResources AzureRM
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ComputeResources-Azure
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+
+from fauxfactory import gen_string
+from nailgun import entities
+from robottelo.api.utils import skip_yum_update_during_provisioning
+from robottelo.config import settings
+from robottelo.constants import (
+    AZURERM_FILE_URI,
+    AZURERM_RHEL7_FT_IMG_URN,
+    AZURERM_RHEL7_UD_IMG_URN,
+    AZURERM_RG_DEFAULT,
+    AZURERM_PREMIUM_OS_Disk,
+    AZURERM_PLATFORM_DEFAULT,
+    AZURERM_VM_SIZE_DEFAULT,
+)
+from robottelo.decorators import (
+    tier1,
+    tier2,
+    tier3,
+    upgrade,
+)
+
+
+class TestAzureRMComputeResourceTestCase:
+    """Tests for ``api/v2/compute_resources``"""
+
+    @upgrade
+    @tier1
+    def test_positive_crud_azurerm_cr(self, module_org, module_location, azurerm_settings):
+        """Create, Read, Update and Delete AzureRM compute resources
+
+        :id: da081a1f-9614-4918-91cb-c900c40ac121
+
+        :expectedresults: Compute resource should be created, read, updated and deleted
+
+        :CaseImportance: Critical
+
+        :CaseLevel: Component
+        """
+
+        # Create CR
+        cr_name = gen_string('alpha')
+        compresource = entities.AzureRMComputeResource(
+            name=cr_name,
+            provider='AzureRm',
+            tenant=azurerm_settings['tenant'],
+            app_ident=azurerm_settings['app_ident'],
+            sub_id=azurerm_settings['sub_id'],
+            secret_key=azurerm_settings['secret'],
+            region=azurerm_settings['region'],
+            organization=[module_org],
+            location=[module_location],
+        ).create()
+        assert compresource.name == cr_name
+        assert compresource.provider == 'AzureRm'
+        assert compresource.tenant == azurerm_settings['tenant']
+        assert compresource.app_ident == azurerm_settings['app_ident']
+        assert compresource.sub_id == azurerm_settings['sub_id']
+        assert compresource.region == azurerm_settings['region']
+
+        # Update CR
+        new_cr_name = gen_string('alpha')
+        description = gen_string('utf8')
+        compresource.name = new_cr_name
+        compresource.description = description
+        compresource.update(['name', 'description'])
```

Comment:
You should assign update result back to variable to ensure that these values are what you expect **after** updating. Otherwise, if update silently fails, your test will still pass.
```suggestion
        compresource = compresource.update(['name', 'description'])
```

Reply:
good catch .. will update it .

Reply:
it is pushed on new commit now.